corehq/form_processor/management/commands/fix_edit_forms_attachments.py

```diff
@@ -0,0 +1,153 @@
+        """
+        find all forms that are a result of a form edition
+        find the deprecated form and then compare attachments between them
+        :return:
+        """
+        add_forms_with_attachments = defaultdict(list)
+        form_accessor = FormAccessorSQL()
+        for db_alias in get_db_aliases_for_partitioned_query():
+            print('scanning %s' % db_alias)
+            edit_form_ids = (XFormOperationSQL.objects.using(db_alias)
+                             .filter(operation='edit')
+                             .values_list('form_id', flat=True).order_by('-date').distinct())
+            for edit_form_id in with_progress_bar(edit_form_ids):
+                if edit_form_id in self.forms:
+                    # this form was already covered by some other form in its hierarchy
+                    continue
+                edit_chain = get_sql_previous_versions(edit_form_id, form_accessor)
+                original_form = edit_chain[0]
+                original_attachments = list(FormAccessorSQL.get_attachments_for_forms([original_form.form_id]))
+                for edited_form in edit_chain[1:]:
+                    attachments = list(FormAccessorSQL.get_attachments_for_forms([edited_form.form_id]))
+                    attachment_names = [a.name for a in attachments]
+                    for attachment in original_attachments:
+                        if attachment.name != 'form.xml' and attachment.name not in attachment_names:
+                            assert attachment.form_id == original_form.form_id
+                            self.forms[edited_form.form_id] = edited_form
+                            for attachment_to_be_added in add_forms_with_attachments[edited_form.form_id]:
+                                if attachment_to_be_added.attachment_id == attachment.attachment_id:
+                                    # this attachment add request is already present
+                                    continue
+                            add_forms_with_attachments[edited_form.form_id].append(attachment)
+        return add_forms_with_attachments
+
+    def _find_couch_forms_with_missing_attachments(self):
+        """
+        find deprecated form ids,
+        then look for the new form and compare attachments between them
+        :return:
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def _add_attachment_to_couch_form(attachment_id, form_id):
+        raise NotImplementedError
+
+    @staticmethod
+    def _add_attachment_to_sql_form(form_id, attachment):
+        new_att = BlobMeta(
+            form_id=form_id,
+            name=attachment.name,
+            attachment_id=uuid.uuid4(),
+            content_type=attachment.content_type,
+            properties=attachment.properties,
+            blob_bucket=attachment.blobdb_bucket(),
+        )
+        with attachment.read_content(stream=True) as content:
+            new_att.write_content(content)
+        new_att.save()
```

Comment:
This entire function body should be replaced with something more like:
```py
# at top of file
from corehq.blobs import CODES, get_blob_db

code = CODES.form_xml if attachment.name == "form.xml" else CODES.form_attachment
get_blob_db().put(
    content,
    domain=attachment.form.domain,  # <--- not sure if this will work, but domain is now required
    parent_id=form_id,
    type_code=code,
    name=attachment.name,
    content_type=attachment.content_type,
    properties=attachment.properties,
)
```

Reply:
thanks @millerdev 
Made the change but have not tested it. Fine not to review till then, will notify once tested.